lib/fluent/plugin/out_sumologic.rb

```diff
@@ -182,15 +182,15 @@ def formatted_to_msgpack_binary
     true
   end
 
-  def sumo_key(sumo_metadata, record, tag)
-    source_name = sumo_metadata['source'] || @source_name
-    source_name = expand_param(source_name, tag, nil, record)
+  def sumo_key(sumo_metadata, tag, chunk)
+    source_name = sumo_metadata['source'] || @source_name || tag.to_s
```

Comment:
Why the logic are inconsistent for `source_name`/`sourcecategory`/`source_host`? 
Is possible`tag` == `nil`?

Reply:
you are right, reverted